packages/fxa-admin-panel/server/lib/server.ts

```diff
@@ -71,13 +81,55 @@ app.get('/__version__', (_, res) =>
   res.type('application/json').send(JSON.stringify(version))
 );
 
+function injectMetaContent(html: string, metaContent: { [x: string]: any }) {
+  let result = html;
+
+  Object.keys(metaContent).forEach(k => {
+    result = result.replace(
+      k,
+      encodeURIComponent(JSON.stringify(metaContent[k]))
+    );
+  });
+
+  return result;
+}
+
+function injectHtmlConfig(
+  html: string,
+  config: { env: string; servers: { admin: { url: any } } }
+) {
+  return injectMetaContent(html, {
+    __SERVER_CONFIG__: config,
+  });
+}
+
 // Note - the static route handlers must come last
 // because the proxyUrl handler's app.use('/') captures
 // all requests that match no others.
 if (proxyUrl) {
   logger.info('static.proxying', { url: proxyUrl });
   const proxy = require('express-http-proxy');
-  app.use('/', proxy(proxyUrl));
+  app.use(
+    '/',
+    proxy(proxyUrl, {
+      userResDecorator: function(
+        proxyRes: { headers: { [x: string]: any } },
+        proxyResData: { toString: (arg0: string) => any },
+        userReq: { url: string } /*, userRes*/
+      ) {
```

Comment:
I don't think we need to declare our own types here for `proxyRes` and `userReq`; they are Request and Response from express.

Shouldn't `toString` always return a string?